### PR TITLE
chore: Convert `InstanceType` to struct

### PR DESCRIPTION
**Pull request by jonathan-innis**
Fixes # 

**Description**

- Completes the migration to modeling `InstanceType` as a struct rather than an interface

Buddy change to https://github.com/aws/karpenter-core/pull/83

**How was this change tested?**

* `make presubmit`
* `FOCUS="KubeletConfiguration" make e2etests`

**Does this change impact docs?**
- [ ] Yes, PR includes docs updates
- [ ] Yes, issue opened: # 
- [x] No

**Release Note**

```release-note
None
```

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: pkg/apis/v1alpha1/register.go [hunk 1]
```diff
@@ -67,6 +67,7 @@ var (
 	LabelInstanceCPU             = LabelDomain + "/instance-cpu"
 	LabelInstanceMemory          = LabelDomain + "/instance-memory"
 	LabelInstancePods            = LabelDomain + "/instance-pods"
+	LabelInstanceMetal           = LabelDomain + "/instance-metal"
```
Comment (ellistarn) [inline pkg/apis/v1alpha1/register.go:70]: rmove

File: website/content/en/preview/tasks/metrics.md [hunk 2]
```diff
@@ -7,7 +7,7 @@ description: >
   Inspect Karpenter Metrics
 ---
 <!-- this document is generated from hack/docs/metrics_gen_docs.go -->
-Karpenter makes several metrics available in Prometheus format to allow monitoring cluster provisioning status. These metrics are available by default at `karpenter.karpenter.svc.cluster.local:8080/metrics` configurable via the `METRICS_PORT` environment variable documented [here](../globalsettings)
+Karpenter makes several metrics available in Prometheus format to allow monitoring cluster provisioning status. These metrics are available by default at `karpenter.karpenter.svc.cluster.local:8080/metrics` configurable via the `METRICS_PORT` environment variable documented [here](../configuration)
```
[the remark this answers is not in the document]
Comment (jonathan-innis) [reply website/content/en/preview/tasks/metrics.md:10]: Nope lol

File: pkg/cloudprovider/instancetypes_test.go [hunk 3]
```diff
@@ -319,10 +314,9 @@ var _ = Describe("Instance Types", func() {
 					},
 				})
 				it := NewInstanceType(ctx, instanceInfo["m5.xlarge"], provisioner.Spec.KubeletConfiguration, "", provider, nil)
-				overhead := it.Overhead()
-				Expect(overhead.Memory().String()).To(Equal("30820Mi"))
```
Comment (ellistarn) [inline pkg/cloudprovider/instancetypes_test.go:323]: Why did this change?

Comment (ellistarn) [reply pkg/cloudprovider/instancetypes_test.go:323]: Do you need a Total here?

**Review by ellistarn — APPROVED** [review]

**Pull request merged**